### ISSUE TITLE
Implement InstanceNorm

### ISSUE THREAD

**Issue by seanpmorgan** [tensorflow/addons#7]
Per the [RFC](https://github.com/tensorflow/community/blob/393dfc9e03dd4150a839de6357a26e0259aa97d6/rfcs/20181214-move-to-addons.md), we need to move [instance_norm](https://github.com/tensorflow/tensorflow/blob/v1.12.0/tensorflow/contrib/layers/python/layers/normalization.py#L42) from contrib to addons:

This will involve restructuring in an OO format by inheriting from base Keras Layer, modifying the code to match those APIs, and modifying test cases to run in TF2.

https://arxiv.org/abs/1607.08022

**Comment by seanpmorgan** [comment]
This will be consolidated in #6
